### PR TITLE
Unified Class and Reference Search 2: Resurrection

### PR DESCRIPTION
**Pull request by YeldhamDev**
That's right, @RayKoopa's [commit](https://github.com/godotengine/godot/pull/14535) has been brought back from Conflict Hell for another chance to be merged! With some changes of mine sprinkled in, such as:

- Make use of `LineEdit`'s own clear button instead of creating a new one.
- Add tooltips.
- TTR the display options.

screenshot at 2018-11-13 00-16-13

### CODE REVIEW

File: core/class_db.cpp [hunk 1]
```diff
@@ -275,7 +275,7 @@ bool ClassDB::is_parent_class(const StringName &p_class, const StringName &p_inh
 
 		if (inherits == p_inherits)
 			return true;
-		inherits = get_parent_class(inherits);
+		inherits = get_parent_class_nocheck(inherits);
```
Comment (akien-mga) [inline core/class_db.cpp]: I'm not sure about this change, the "check" version was probably used for a reason. Might be needed to add `is_parent_class_nocheck` instead? But maybe let's see what @reduz thinks about it.

**Comment by RayKoopa** [comment]
I think something was redundant due to my own check above. But yeah, it's been almost a year.

**Comment by RayKoopa** [comment]
I actually don't remember anymore. It's been a year. Maybe related to https://github.com/godotengine/godot/issues/14872 ?

File: core/class_db.cpp (hunk 1 above)
Comment (RayKoopa) [reply core/class_db.cpp]: Well I probably had a reason to put it in there. Maybe the check was redundant?

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**